crates/datafusion_ext/src/vars.rs

```diff
@@ -95,6 +95,7 @@ impl SessionVars {
      is_cloud_instance: bool,
      dialect: Dialect,
      enable_experimental_scheduler: bool,
+     prefer_local_execution: bool,
```

Comment:
- is this really "prefer" and not "force"
- are there meaningfully cases in local/embedded nodes where there's more-than-one session? My instinct was that local sessions (which are the things where hybrid execution could have happened anyway,) were always single user, and had one copy of the glaredb instance loaded/running. 
- seems like we should (maybe) have this be an option on the operation itself rather than on the session (which means that the state of "should this operation be remote or local" is stateful on the session, and could be weird in multi-threaded contexts.
-

Reply:
`prefer_local_execution` might not be the best name for this, but it's really more about setting the default location of where to run parts of the query. Currently we default to "remote" at the root and selectively pull up "local" parts as much as possible. This variable really just changes it to default "local" at the root.

I suggested a session variable since changing the default _could_ break something that we don't know about, and so having a way to go back to the other behavior seemed reasonable and lets us play with it.

Reply:
can we just have this be "execution_mode" and have it be an enum?

`should_hybrid_execution_default_local` 

------

if it defaults to local at the root will any parts be remote?

Reply:
I am working on the same, as Sean mentioned. Right now, the "remote" isn't pulled up; it must also be configurable. I will mark it ready for review once it is done.

Reply:
[sounds good to me! sorry if I complicated things, but this feels clearer!]

Reply:
> I am working on the same, as Sean mentioned. Right now, the "remote" isn't pulled up; it must also be configurable. I will mark it ready for review once it is done.

Oh, I assumed it was already doing that. If we can have it pull up matching preferences, that would be ideal (and would be useful once extended for dist exec too).